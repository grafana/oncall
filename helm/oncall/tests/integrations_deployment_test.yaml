suite: test integrations deployment
templates:
  - integrations/deployment.yaml
release:
  name: oncall
chart:
  appVersion: v1.3.39
tests:
  - it: detached_integrations.enabled=false -> should not create deployment (default)
    asserts:
      - hasDocuments:
          count: 0

  - it: detached_integrations.enabled=true -> should create integrations deployment
    set:
      detached_integrations.enabled: true
    asserts:
      - containsDocument:
          kind: Deployment
          apiVersion: apps/v1
          metadata.name: oncall-integrations
      - isSubset:
          path: metadata.labels
          content:
            app.kubernetes.io/component: integrations
            app.kubernetes.io/instance: oncall
            app.kubernetes.io/name: oncall
      - isSubset:
          path: spec.selector.matchLabels
          content:
            app.kubernetes.io/component: integrations
            app.kubernetes.io/instance: oncall
            app.kubernetes.io/name: oncall
      - isSubset:
          path: spec.template.metadata.labels
          content:
            app.kubernetes.io/component: integrations
            app.kubernetes.io/instance: oncall
            app.kubernetes.io/name: oncall
      - equal:
          path: spec.replicas
          value: 1
      - equal:
          path: spec.template.spec.serviceAccountName
          value: oncall
      - contains:
          path: spec.template.spec.initContainers
          content:
            name: wait-for-db
          any: true
      - matchSnapshot:
          path: spec.template.spec.containers

---
suite: test integrations internal service
templates:
  - integrations/service-internal.yaml
release:
  name: oncall
chart:
  appVersion: v1.3.39
tests:
  - it: detached_integrations.enabled=false -> should not create service
    asserts:
      - hasDocuments:
          count: 0

  - it: detached_integrations.enabled=true -> should create service
    set:
      detached_integrations.enabled: true
    asserts:
      - containsDocument:
          kind: Service
          apiVersion: v1
          metadata.name: oncall-integrations
      - matchSnapshot:
          path: spec
