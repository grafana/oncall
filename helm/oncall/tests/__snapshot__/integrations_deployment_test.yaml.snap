detached_integrations.enabled=true -> should create integrations deployment:
  1: |
    - env:
        - name: BASE_URL
          value: https://example.com
        - name: SECRET_KEY
          valueFrom:
            secretKeyRef:
              key: SECRET_KEY
              name: oncall
        - name: MIRAGE_SECRET_KEY
          valueFrom:
            secretKeyRef:
              key: MIRAGE_SECRET_KEY
              name: oncall
        - name: MIRAGE_CIPHER_IV
          value: 1234567890abcdef
        - name: DJANGO_SETTINGS_MODULE
          value: settings.helm
        - name: AMIXR_DJANGO_ADMIN_PATH
          value: admin
        - name: OSS
          value: "True"
        - name: DETACHED_INTEGRATIONS_SERVER
          value: "True"
        - name: UWSGI_LISTEN
          value: "1024"
        - name: BROKER_TYPE
          value: rabbitmq
        - name: GRAFANA_API_URL
          value: http://oncall-grafana
        - name: FEATURE_SLACK_INTEGRATION_ENABLED
          value: "False"
        - name: FEATURE_TELEGRAM_INTEGRATION_ENABLED
          value: "False"
        - name: FEATURE_EMAIL_INTEGRATION_ENABLED
          value: "True"
        - name: EMAIL_HOST
          value: null
        - name: EMAIL_PORT
          value: "587"
        - name: EMAIL_HOST_USER
          value: null
        - name: EMAIL_HOST_PASSWORD
          valueFrom:
            secretKeyRef:
              key: smtp-password
              name: oncall-smtp
              optional: true
        - name: EMAIL_USE_TLS
          value: "True"
        - name: EMAIL_USE_SSL
          value: "False"
        - name: EMAIL_FROM_ADDRESS
          value: null
        - name: EMAIL_NOTIFICATIONS_LIMIT
          value: "200"
        - name: FEATURE_PROMETHEUS_EXPORTER_ENABLED
          value: "False"
        - name: MYSQL_HOST
          value: oncall-mariadb
        - name: MYSQL_PORT
          value: "3306"
        - name: MYSQL_DB_NAME
          value: oncall
        - name: MYSQL_USER
          value: root
        - name: MYSQL_PASSWORD
          valueFrom:
            secretKeyRef:
              key: mariadb-root-password
              name: oncall-mariadb
        - name: REDIS_PROTOCOL
          value: redis
        - name: REDIS_HOST
          value: oncall-redis-master
        - name: REDIS_PORT
          value: "6379"
        - name: REDIS_DATABASE
          value: "0"
        - name: REDIS_USERNAME
          value: ""
        - name: REDIS_PASSWORD
          valueFrom:
            secretKeyRef:
              key: redis-password
              name: oncall-redis
        - name: RABBITMQ_USERNAME
          value: user
        - name: RABBITMQ_PASSWORD
          valueFrom:
            secretKeyRef:
              key: rabbitmq-password
              name: oncall-rabbitmq
        - name: RABBITMQ_HOST
          value: oncall-rabbitmq
        - name: RABBITMQ_PORT
          value: "5672"
        - name: RABBITMQ_PROTOCOL
          value: amqp
        - name: RABBITMQ_VHOST
          value: ""
        - name: ROOT_URLCONF
          value: engine.integrations_urls
      image: grafana/oncall:v1.3.39
      imagePullPolicy: Always
      livenessProbe:
        httpGet:
          path: /health/
          port: http
        periodSeconds: 60
        timeoutSeconds: 3
      name: oncall
      ports:
        - containerPort: 8080
          name: http
          protocol: TCP
      readinessProbe:
        httpGet:
          path: /ready/
          port: http
        periodSeconds: 60
        timeoutSeconds: 3
      resources: {}
      securityContext: {}
      startupProbe:
        httpGet:
          path: /startupprobe/
          port: http
        periodSeconds: 10
        timeoutSeconds: 3
